crates/polars-core/src/chunked_array/logical/time.rs

```diff
@@ -10,8 +13,47 @@ impl From<Int64Chunked> for TimeChunked {
 }
 
 impl Int64Chunked {
-    pub fn into_time(self) -> TimeChunked {
-        TimeChunked::new_logical(self)
+    pub fn into_time(mut self) -> TimeChunked {
+        let mut null_count = 0;
+
+        let chunks = std::mem::take(&mut self.chunks)
+            .into_iter()
+            .map(|chunk| {
```

Comment:
I think we can use `to_arrow` which gives us the logical arrow type and then call the arrow `cast` kernels. Those should do these checks.

Reply:
The `polars_arrow::compute::cast::cast` does not do these checks, if you are referring to that.

Reply:
Then that kernel should.

Reply:
Okay, I will fix that then.